pkg/apiserver/resthandler.go

```diff
@@ -753,7 +781,7 @@ func DeleteResource(r rest.GracefulDeleter, checkBody bool, scope RequestScope,
 		if admit != nil && admit.Handles(admission.Delete) {
 			userInfo, _ := api.UserFrom(ctx)
 
-			err = admit.Admit(admission.NewAttributesRecord(nil, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Delete, userInfo))
+			err = admit.Admit(admission.NewAttributesRecord(nil, nil, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Delete, userInfo))
```

Comment:
Additional state semantics around deletion - reject if controller still
exists, for example.

On May 23, 2016, at 8:17 PM, Chao Xu <notifications@github.com> wrote:

In pkg/apiserver/resthandler.go
<https://github.com/kubernetes/kubernetes/pull/25787#discussion_r64311017>:

> @@ -753,7 +781,7 @@ func DeleteResource(r rest.GracefulDeleter, checkBody bool, scope RequestScope,
>  		if admit != nil && admit.Handles(admission.Delete) {
>  			userInfo, _ := api.UserFrom(ctx)
>
> -			err = admit.Admit(admission.NewAttributesRecord(nil, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Delete, userInfo))
> +			err = admit.Admit(admission.NewAttributesRecord(nil, nil, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Delete, userInfo))

Could you give an example on why knowing the oldObj is useful for the
admission a deletion?

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/25787/files/bf47e5790193381752c396876a2ce9950fe01e6d#r64311017>